### PR TITLE
Fix 3619. Bump dependency catch2

### PR DESCRIPTION
**Pull request by LukashonakV**
Signed-off-by: Viktar Lukashonak <myxabeer@gmail.com>

### CODE REVIEW

**Comment by LukashonakV** [comment]
Hi @Alexays, @alebastr . Waybar meson configuration does not aware off whether tests option enabled or not. And In case when catch2 library is found forces to compile test subproject. I've reproduces case is mentioned in #3619 . By default catch2 has got 'tests' option disabled. In my opinion in such case no need to compile test subproject.

File: meson.build [hunk 1]
```diff
@@ -586,11 +586,12 @@ endif
 
 catch2 = dependency(
     'catch2',
+    version: '>=3.7.1',
     default_options: [ 'tests=false' ],
     fallback: ['catch2', 'catch2_dep'],
     required: get_option('tests'),
 )
-if catch2.found()
+if catch2.found() and get_option('tests').enabled()
```
Comment (alebastr) [inline meson.build:594]: This isn't correct, please see <https://mesonbuild.com/Reference-manual_functions.html#dependency_required>. `-Dtests=disabled` already skips the dependency lookup and sets `catch2` to an empty dependency object.
```
Dependency catch2 skipped: feature tests disabled
```

Also, no reason to enforce the version requirement. Last time I checked the test code did not use anything specific to catch2 version 3.7.1 and worked with version 2.x.

Comment (LukashonakV) [reply meson.build:594]: Hi @alebastr , very interesting case:
1. Actually by default meson configuration for waybar provides two or even more features with the same name ```tests```. One of belongs to the project itself and others one to their dependencies. In my case ```tests``` feature is provided by the gtk-layer-shell-0 and catch2 . So when catch2 dependency is not met in the system, meson trough dependency fallback provides ```tests=false```
2. When catch2 dependency is presented in the system then ```tests``` project feature is used. In my system by default ```tests=auto``` which leads to test subproject compilation. In this case for some reason linker failes with the undefined references
----
Need to digging deeper to check why linker fails

Comment (LukashonakV) [reply meson.build:594]: Looks like catch2 issue. I create pure small project with catch2 dependency only and not able to compile it. Will create issue to catch2

Comment (LukashonakV) [reply meson.build:594]: Appropriate issue is created catchorg/Catch2#2912

**Pull request closed without merging**